### PR TITLE
ci: enable govet and gosimple linters, fix found bugs

### PR DESCRIPTION
**Pull request by kolyshkin**
Found/fixed 3 issues with govet and 3 issues with gosimple.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@nalind @giuseppe @vrothberg @mtrmac @umohnani8 @saschagrunert PTAL

**Review by saschagrunert — APPROVED** [review]

**Comment by nalind** [comment]
LGTM.

File: pkg/regexp/regexp.go [hunk 1]
```diff
@@ -22,7 +22,10 @@ func Delayed(val string) Regexp {
 	}
 	if precompile {
 		re.regexp = regexp.MustCompile(re.val)
+	} else {
+		re.once = &sync.Once{}
 	}
+
```
Comment (mtrmac) [inline pkg/regexp/regexp.go:28]: Having multiple `Regexp` values pointing to a shared `once` doesn’t make sense.

Instead, make `type Regexp` a `*struct`, and avoid the copy that way.

Comment (mtrmac) [reply pkg/regexp/regexp.go:28]: Filed #1636 .

**Pull request merged**